x-pack/plugins/ml/common/types/custom_urls.ts

```diff
@@ -19,9 +20,22 @@ export interface KibanaUrlConfig extends BaseUrlConfig {
   time_range?: string;
 }
 
+export interface KibanaUrlConfigWithTimeRange extends BaseUrlConfig {
+  time_range: string;
+}
+
 export type UrlConfig = BaseUrlConfig | KibanaUrlConfig;
 
 export interface CustomUrlAnomalyRecordDoc extends AnomalyRecordDoc {
   earliest: string;
   latest: string;
 }
+
+export function isKibanaUrlConfigWithTimeRange(arg: unknown): arg is KibanaUrlConfigWithTimeRange {
+  return (
+    isPopulatedObject(arg) &&
```

Comment:
Nit: You could move part of the check to isPopulatedObject here like this:
```
isPopulatedObject(arg, ['url_name', 'url_value', 'time_range']) && typeof arg.time_range === 'string'
```

Reply:
Updated in 7ee455c02c2252a159431305bf8b90cabaed6305